src/Core/Image/ImageFormatConfiguration.php

```diff
@@ -51,7 +57,30 @@ public function __construct(ConfigurationInterface $configuration)
 
     public function getGenerationFormats(): array
     {
-        return explode(self::SEPARATOR, $this->configuration->get(self::IMAGE_FORMAT_CONFIGURATION_KEY));
+        // Return formats from cache
+        if (!empty($this->formatsToGenerate)) {
+            return $this->formatsToGenerate;
+        }
+
+        // We will start with the base format, that will be generated no matter what
+        $this->formatsToGenerate = [self::DEFAULT_IMAGE_FORMAT];
+
+        // If multiple image formats feature is not enabled, we return only the fallback format
+        if (!FeatureFlag::isEnabled(FeatureFlagSettings::FEATURE_FLAG_MULTIPLE_IMAGE_FORMAT)) {
```

Comment:
I don't agree with this in develop branch. We must use FeatureFlagStateCheckerInterface instead!

Reply:
The thing is this class is also used in legacy context and in FO especially in `ImageRetriever` (I didn't look into all usages but ther's at least this one)

So we have no alternative but to use legacy ObjectModel in this case, the proper alternative would be to make the `FeatureFlagStateCheckerInterface` accessible in FO as well, but for now it's not possible and it's not the scope of a merge PR to add this

So I would suggest in this case creating a refacto issue to make this possible, or even better we should finish this feature and we wouldn't even need to check the feature flag anymore But since this feature is not exactly in top priority, chances are the refacto solution is probably the most realistic one

Reply:
@boherm @jolelievre I had to do it like this, because it did not work in FO and this was the easiest thing to do.